eth_account/account.py

```diff
@@ -1043,3 +1054,99 @@ def sign_typed_data(
         )
         message_hash = _hash_eip191_message(signable_message)
         return cast(SignedMessage, self._sign_hash(message_hash, private_key))
+
+    @combomethod
+    def sign_authorization(
+        self,
+        authorization_dict: AuthorizationDict,
+        private_key: PrivateKeyType,
+    ) -> SignedSetCodeAuthorization:
+        r"""
+        Sign an authorization using a local private key to be included in a EIP-7702 transaction.
+        It adds the signature fields to the authorization dict.
+
+        :param dict authorization_dict: the required keys are: chainId, address, nonce
+        :param private_key: the private key to sign the data with
+        :type private_key: hex str, bytes, int or :class:`eth_keys.datatypes.PrivateKey`
+        :returns: the dictionary with the signature fields added, suitable for inclusion in a EIP-7702 transaction
+        :rtype: dict
+
+        .. NOTE::
+            You need to get signed one or more signed authorizations from an EOA willing to have a smart contract code associated with the EOA, this the essence of EIP-7702
+
+            in the code below in the variable authorization_to_sign:
```

Comment:
```suggestion
            in the code below in the variable `auth``:
```

Reply:
this is `rst`, so I think you'll want ``auth``. Same with the below